File: src/main/java/com/fish/extendedae_plus_client/lang/EAEPCDataGenerators.java
package com.fish.extendedae_plus_client.lang;

import com.fish.extendedae_plus_client.ExtendedAEPlusClient;
import net.neoforged.bus.api.SubscribeEvent;
import net.neoforged.fml.common.EventBusSubscriber;
import net.neoforged.neoforge.data.event.GatherDataEvent;

@EventBusSubscriber(modid = ExtendedAEPlusClient.MODID)
public class EAEPCDataGenerators {
    @SubscribeEvent
    private static void register(GatherDataEvent event) {
        var generator = event.getGenerator();
        var output = generator.getPackOutput();

        generator.addProvider(event.includeClient(), new LangEN(output));
        generator.addProvider(event.includeClient(), new LangZH(output));
    }
}

File: src/main/kotlin/com/fish/extendedae_plus_client/lang/EAEPCDataGenerators.kt
package com.fish.extendedae_plus_client.lang

import com.fish.extendedae_plus_client.ExtendedAEPlusClient
import net.minecraftforge.data.event.GatherDataEvent
import net.minecraftforge.eventbus.api.SubscribeEvent
import net.minecraftforge.fml.common.Mod

@Mod.EventBusSubscriber(modid = ExtendedAEPlusClient.MODID)
object EAEPCDataGenerators {
    @SubscribeEvent
    fun register(event: GatherDataEvent) {
        val generator = event.generator
        val output = generator.packOutput

        generator.addProvider(event.includeClient(), LangEN(output))
        generator.addProvider(event.includeClient(), LangZH(output))
    }
}
